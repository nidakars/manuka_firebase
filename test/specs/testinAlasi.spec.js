import { mount, RouterLinkStub } from "@vue/test-utils";

import ustMenu from "../../components/mainMenu";

describe("mainMenu.vue", () => {
    let wrapper = mount(ustMenu, {
        stubs: {
            NuxtLink: RouterLinkStub
        }
    });
    it("BaÅŸlÄ±k 1", () => {
        expect(wrapper.text()).toContain("Moonlight KoleksiyonuðŸŒ™");
    });
    it("BaÅŸlÄ±k 2", () => {
        expect(wrapper.text()).toContain("Denim Koleksiyonu");
    });
    it("BaÅŸlÄ±k 3", () => {
        expect(wrapper.text()).toContain("Makaron Renkleri");
    });
    it("BaÅŸlÄ±k 4", () => {
        expect(wrapper.text()).toContain("AvantajlÄ± Fiyatlar");
    });
    it("BaÅŸlÄ±k 5", () => {
        expect(wrapper.text()).toContain("Ã‡ok Satanlar");
    });
    it("Ana BaÅŸlÄ±k", () => {
        expect(wrapper.text()).toContain("Yeni");
    });
    it("div'e gÃ¶re", () => {
        const div = wrapper.find('div')
        expect(div.exists()).toBe(true)
    });

    test("snapshot testi", () => {
        expect(wrapper).toMatchSnapshot();
    });
    it("button var mÄ±?", () => {
        expect(wrapper.contains("button")).toBe(true);
    });
});