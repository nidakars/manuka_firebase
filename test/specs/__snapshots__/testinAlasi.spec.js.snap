// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mainMenu.vue snapshot testi 1`] = `
<header>
  <div id="header-wrap" class="fl col-12">
    <div class="col-12 HeaderInner">
      <div id="headerMain" class="col col-12 flexwrap">
        <div class="fl col-md-1 col-sm-2 forMobile menuBtnBox"><span id="menuBtn"></span></div>
        <div id="logo" class="col col-3 col-sm-4 ease"><a href="/">
            <p><img src="https://img.manuka.com.tr/Data/EditorFiles/tema/img/manuka-logo-yeni.png" alt="manuka.com.tr"></p>
          </a></div>
        <div class="col-4 PcSearch forDesktop">
          <div id="search" class="fr col-12 ease">
            <form id="FormAra" name="FormAra" action="/arama" method="get" data-before="Ara" class="icon-no-space"><input id="live-search" autocomplete="off" name="q" type="text" placeholder="Ne AramÄ±ÅŸtÄ±nÄ±z ?" content="text/html; charset=UTF-8" class="col withPlace ease personaclick-instant-search personaclick-instant-search-rendered"> <button type="submit" value="Ara" id="search-btn" class="fr ease"></button>
              <div id="live-search-box" data-callback="searchCallback" class="pos-a col-12 whiteBg box-border" style="visibility: hidden;">
                <div class="box col-12 p-top">
                  <ul ng-if="products.length > 0" class="box col-12 p-bottom">
                    <li class="col col-12 text-title text-bold text-upper line-bottom mb"><span class="row"><i class="icon-cube"></i>
                          ÃœrÃ¼nler
                        </span></li>
                    <li ng-repeat="product in products" class="col col-12"><a class="col col-12 text-description mb"></a></li>
                  </ul>
                  <ul ng-if="categories.length > 0" class="box col-12 p-bottom">
                    <li class="col col-12 text-title text-bold text-upper line-bottom mb"><span class="row"><i class="icon-sitemap"></i>
                          Kategoriler
                        </span></li>
                    <li ng-repeat="category in categories" class="col col-12"><a class="col col-12 text-description mb"></a></li>
                  </ul>
                  <ul ng-if="brands.length > 0" class="box col-12 p-bottom">
                    <li class="col col-12 text-title text-bold text-upper line-bottom mb"><span class="row"><i class="icon-building"></i>
                          Markalar
                        </span></li>
                    <li ng-repeat="brand in brands" class="col col-12"><a class="col col-12 text-description mb"></a></li>
                  </ul>
                </div>
              </div>
            </form>
          </div>
        </div>
        <div id="headerRight" class="fr col p-right col-4 col-md-8 col-sm-6 flexwrap"><a data-width="500" data-url="/srv/service/customer/login-form" class="fr text-center authorLink popupWin"></a> <a data-width="500" data-url="/srv/service/customer/login-form" id="favoribtn" class="col fr popupWin"><span class="favSayi"></span></a> <a id="cart-soft-count" class="fr basketLink"><span class="cart-soft-count"> function () { [native code] }</span></a></div>
      </div>
      <nav id="mainMenu" class="fl col-12 forDesktop">
        <ul class="fl col-12 menu flexwrap">
          <li class="fl"><span class="fl d-flex drop-down-title"><a lang="tr" href="/indirim" class="fl">Ä°ndirim</a></span></li>
          <li class="fl"><span class="fl d-flex drop-down-title"><a lang="tr" href="/yeni" class="fl">Yeni</a></span></li>
          <li class="fl drop-down hover parentLink"><span class="fl d-flex drop-down-title"><a lang="tr" class="fl">Giyim</a></span>
            <div class="pos-a pos-top pos-left pos-right box p-left whiteBg box-border b-top subMenu top">
              <div class="inner">
                <div class="box col-12 dynamicMenu">
                  <ul class="fl col-3 SubNewCat">
                    <li class="fl col-12"><a href="/moonlight" class="fl col-12">Moonlight KoleksiyonuðŸŒ™</a></li>
                    <li class="fl col-12"><a href="/naturel-koleksiyonu" class="fl col-12">NatÃ¼rel Koleksiyonu</a></li>
                    <li class="fl col-12"><a href="/denim" class="fl col-12">Denim Koleksiyonu</a></li>
                    <li class="fl col-12"><a href="/makaron-renkleri" class="fl col-12">Makaron Renkleri</a></li>
                    <li class="fl col-12"><a href="/avantajli-fiyatlar" class="fl col-12 MenuRed">AvantajlÄ± Fiyatlar</a></li>
                    <li class="fl col-12"><a href="/cok-satanlar" class="fl col-12">Ã‡ok Satanlar</a></li>
                  </ul>
                  <div class="col-9 col p-right SubMenuRight flexwrap">
                    <ul class="col p-left col-3">
                      <li class="fl col-12"><a href="/kaban-333" class="fl col-12">Kaban</a></li>
                      <li class="fl col-12"><a href="/mont-334" class="fl col-12">Mont</a></li>
                      <li class="fl col-12"><a href="/yelek" class="fl col-12">Yelek</a></li>
                      <li class="fl col-12"><a href="/yagmurluk-327" class="fl col-12">YaÄŸmurluk</a></li>
                      <li class="fl col-12"><a href="/trenchkot" class="fl col-12">TrenÃ§kot</a></li>
                      <li class="fl col-12"><a href="/denim" class="fl col-12">Denim</a></li>
                      <li class="fl col-12"><a href="/ceket-339" class="fl col-12">Ceket</a></li>
                      <li class="fl col-12"><a href="/kimono" class="fl col-12">Kimono</a></li>
                      <li class="fl col-12"><a href="/triko-325" class="fl col-12">Triko</a></li>
                      <li class="fl col-12"><a href="/hirka-332" class="fl col-12">HÄ±rka</a></li>
                      <li class="fl col-12"><a href="/kazak-323" class="fl col-12">Kazak</a></li>
                      <li class="fl col-12"><a href="/suveter" class="fl col-12">SÃ¼veter</a></li>
                      <li class="fl col-12"><a href="/sweatshirt-338" class="fl col-12">Sweatshirt</a></li>
                      <li class="fl col-12"><a href="/gomlek-329" class="fl col-12">GÃ¶mlek</a></li>
                      <li class="fl col-12"><a href="/tunik" class="fl col-12">Tunik</a></li>
                      <li class="fl col-12"><a href="/basic-328" class="fl col-12">Basic</a></li>
                      <li class="fl col-12"><a href="/t-shirt-324" class="fl col-12">T-Shirt</a></li>
                      <li class="fl col-12"><a href="/elbise-322" class="fl col-12">Elbise</a></li>
                      <li class="fl col-12"><a href="/salopet" class="fl col-12">Salopet</a></li>
                      <li class="fl col-12"><a href="/sweatshirt-takim" class="fl col-12">Sweatshirt TakÄ±m</a></li>
                      <li class="fl col-12"><a href="/esofman-takim" class="fl col-12">EÅŸofman TakÄ±m</a></li>
                      <li class="fl col-12"><a href="/takim-331" class="fl col-12">TakÄ±m</a></li>
                      <li class="fl col-12"><a href="/jean-336" class="fl col-12">Jean</a></li>
                      <li class="fl col-12"><a href="/pantolon-335" class="fl col-12">Pantolon</a></li>
                      <li class="fl col-12"><a href="/etek-337" class="fl col-12">Etek</a></li>
                    </ul>
                    <ul class="col p-left col-3">
                      <li class="MenuSubBaslik">Edit</li>
                      <li class="fl col-12"><a href="/denim-zamani" class="fl col-12">Denim ZamanÄ±<sup style="color: black;">Yeni</sup></a></li>
                    </ul>
                    <div class="col p-left col-6"><img src="https://img.manuka.com.tr/Data/EditorFiles/tema/img/giyimbolumu.jpg"></div>
                  </div>
                </div>
              </div>
            </div>
          </li>
          <li class="fl drop-down hover parentLink"><span class="fl d-flex drop-down-title"><a lang="tr" href="/#/canta" class="fl">Ã‡anta</a></span>
            <div class="pos-a pos-top pos-left pos-right box p-left whiteBg box-border b-top subMenu top">
              <div class="inner">
                <div class="box col-12 dynamicMenu p-left">
                  <ul class="col col-12">
                    <li class="fl col-12"><a href="/everyday-bag" class="fl col-12">Everyday Bag</a></li>
                    <li class="fl col-12"><a href="/bez-canta" class="fl col-12">Kanvas Ã‡anta</a></li>
                    <li class="fl col-12"><a href="/deri-canta" class="fl col-12">Deri Ã‡anta</a></li>
                    <li class="fl col-12"><a href="/canta-340" class="fl col-12">Bez Ã‡anta</a></li>
                    <li class="fl col-12"><a href="/sirt-cantasi" class="fl col-12">SÄ±rt Ã‡antasÄ±</a></li>
                    <li class="fl col-12"><a href="/suni-kurk" class="fl col-12">Suni KÃ¼rk Ã‡anta</a></li>
                    <li class="fl col-12"><a href="/cuzdan" class="fl col-12">CÃ¼zdan</a></li>
                  </ul>
                </div>
              </div>
            </div>
          </li>
          <li class="fl drop-down hover parentLink"><span class="fl d-flex drop-down-title"><a lang="tr" href="/#/aksesuar" class="fl">Aksesuar</a></span>
            <div class="pos-a pos-top pos-left pos-right box p-left whiteBg box-border b-top subMenu top">
              <div class="inner">
                <div class="box col-12 dynamicMenu p-left">
                  <ul class="col col-12">
                    <li class="fl col-12"><a href="/sapka" class="fl col-12">Åžapka</a></li>
                    <li class="fl col-12"><a href="/telefon-kilifi-366" class="fl col-12">Telefon KÄ±lÄ±fÄ±</a></li>
                    <li class="fl col-12"><a href="/defter" class="fl col-12">Defter</a></li>
                    <li class="fl col-12"><a href="/bileklik" class="fl col-12">Bileklik</a></li>
                    <li class="fl col-12"><a href="/yaka" class="fl col-12">Yaka</a></li>
                    <li class="fl col-12"><a href="/kolye-320" class="fl col-12">Kolye</a></li>
                    <li class="fl col-12"><a href="/kemer-321" class="fl col-12">Kemer</a></li>
                  </ul>
                </div>
              </div>
            </div>
          </li>
          <li class="fl drop-down hover parentLink"><span class="fl d-flex drop-down-title"><a lang="tr" href="/#/ayakkabi" class="fl">AyakkabÄ±</a></span>
            <div class="pos-a pos-top pos-left pos-right box p-left whiteBg box-border b-top subMenu top">
              <div class="inner">
                <div class="box col-12 dynamicMenu p-left">
                  <ul class="col col-12">
                    <li class="fl col-12"><a href="/postal-341" class="fl col-12">Postal</a></li>
                    <li class="fl col-12"><a href="/snikers" class="fl col-12">Sneakers</a></li>
                  </ul>
                </div>
              </div>
            </div>
          </li>
          <li class="fl drop-down hover parentLink"><span class="fl d-flex drop-down-title"><a lang="tr" href="/#/corap" class="fl">Ã‡orap</a></span>
            <div class="pos-a pos-top pos-left pos-right box p-left whiteBg box-border b-top subMenu top">
              <div class="inner">
                <div class="box col-12 dynamicMenu">
                  <div class="col-9 col p-right SubMenuRight flexwrap">
                    <ul class="col p-left col-3">
                      <li class="fl col-12"><a href="/kutulu-corap" class="fl col-12">Kutu Ã‡orap</a></li>
                    </ul>
                    <ul class="col p-left col-3">
                      <li class="fl col-12"><a href="/desenli-soket-corap" class="fl col-12">Desenli Ã‡orap</a></li>
                    </ul>
                    <ul class="col p-left col-3">
                      <li class="fl col-12"><a href="/duz-soket-corap" class="fl col-12">DÃ¼z Soket Ã‡orap</a></li>
                    </ul>
                    <ul class="col p-left col-3">
                      <li class="fl col-12"><a href="/fitilli-soket-corap" class="fl col-12">Fitilli Soket Ã‡orap</a></li>
                    </ul>
                  </div>
                </div>
              </div>
            </div>
          </li>
        </ul>
      </nav>
      <div class="col col-12">
        <div id="mobileSearch" class="fl col-12"></div>
      </div>
    </div>
  </div>
</header>
`;
